vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/ComboBoxDataController.java

```diff
@@ -284,6 +286,8 @@ void setRequestedRange(int start, int length, String filter) {
         }
         dataCommunicator.setRequestedRange(start, length);
         filterSlot.accept(filter);
+        forceResetDueToClientSideFilterChange = Objects.equals(filter,
```

Comment:
The approach is currently based on an assumption about the timing and order in which the connector requests data. I wonder if there is a way to avoid an inter-request flag and rather limit the fix to the `resetDataCommunicator` method only.

Reply:
Good point. Updated the logic to use a flag that is updated on data communicator reset. This way, whenever it is called should not matter anymore.